File: BE/avalon/app/schemas/consultation.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class ConsultationBase(BaseModel):
    student_id: int
    semester_id: int
    consultant_id: int
    consultation_name: Optional[str] = None
    consultation_detail: Optional[str] = None
    is_deleted: bool = False

class ConsultationCreate(ConsultationBase):
    pass

class ConsultationUpdate(BaseModel):
    consultation_name: Optional[str] = None
    consultation_detail: Optional[str] = None
    is_deleted: Optional[bool] = None

class Consultation(ConsultationBase):
    consultation_id: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: FE/AL-homepage/src/types/consultation.ts
export interface Consultation {
  consultation_id: number
  student_id: number
  semester_id: number
  consultant_id: number
  consultation_name: string
  consultation_detail: string
  is_deleted: boolean
  created_at: string // ISO 8601 date string
  updated_at: string | null // ISO 8601 date string or null
}

export interface CreateConsultationRequest {
  student_id: number
  semester_id: number
  consultant_id: number
  consultation_name: string
  consultation_detail: string
}

export interface UpdateConsultationRequest {
  consultation_name?: string
  consultation_detail?: string
}
